builtin/credential/approle/validation.go

```diff
@@ -118,6 +118,11 @@ func decodeSecretIDStorageEntry(entry *logical.StorageEntry) (*secretIDStorageEn
 	}
 
 	cleanup := func(in []string) []string {
+		if len(in) == 0 {
+			// Don't change unnecessarily, if it was empty list leave as empty list
+			// instead of making it nil.
+			return in
```

Comment:
Wonder if we should have test for this?

Reply:
Feels like one of those things where if we were to apply a consistent standard, we'd need to add thousands more tests.

Reply:
Yeah, I figured as much. All good.